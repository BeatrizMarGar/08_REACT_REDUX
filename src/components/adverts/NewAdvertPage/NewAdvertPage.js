import React from 'react';
import T from 'prop-types';
import Layout from '../../layout';
import NewAdvertForm from './NewAdvertForm';
import { createAd } from '../../../store/actions';
import useStoreAction from '../../../hooks/StoreActions';

function NewAdvertPage() {
  const createnewad = useStoreAction(createAd)

  return (
    <Layout>
      <NewAdvertForm onSubmit={createnewad} />
    </Layout>
  );
}

export default NewAdvertPage;
