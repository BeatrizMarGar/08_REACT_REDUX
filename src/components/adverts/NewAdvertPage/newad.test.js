
import NewAdvertPage from './NewAdvertPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('NewAdvertPage', () => {
  test('snapshot test', () => {
    const store = {
      getState: () => ({ auth: true, tags: ['motor', 'mobile', 'lifestyle', 'work'] }),
      subscribe: () => {},
      dispatch: () => {},
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NewAdvertPage />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
