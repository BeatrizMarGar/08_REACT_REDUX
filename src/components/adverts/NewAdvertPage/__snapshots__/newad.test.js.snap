// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAdvertForm snapshot 1`] = `
<div>
  <form>
    <input
      name="name"
      value=""
    />
    <input
      checked=""
      name="sale"
      type="checkbox"
    />
    <input
      name="price"
      type="number"
      value="0"
    />
    <div>
      <label>
        <input
          name="tags"
          type="checkbox"
          value="motor"
        />
        motor
      </label>
      <label>
        <input
          name="tags"
          type="checkbox"
          value="mobile"
        />
        mobile
      </label>
      <label>
        <input
          name="tags"
          type="checkbox"
          value="lifestyle"
        />
        lifestyle
      </label>
      <label>
        <input
          name="tags"
          type="checkbox"
          value="work"
        />
        work
      </label>
    </div>
    <input
      name="photo"
      style="display: none;"
      type="file"
    />
    <img
      alt=""
      height="200"
      src="placeholder.png"
      style="object-fit: contain;"
      width="200"
    />
    <button
      disabled=""
    >
      Save
    </button>
  </form>
  ;
</div>
`;

exports[`NewAdvertPage snapshot test 1`] = `
Array [
  <header>
    Nodepop React
    <nav>
      <ul>
        <li>
          <a
            aria-current={null}
            href="/adverts"
            onClick={[Function]}
          >
            Nodepop
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/adverts/new"
            onClick={[Function]}
          >
            New advert
          </a>
        </li>
      </ul>
    </nav>
    <button
      onClick={[Function]}
    >
      Desconectar
    </button>
  </header>,
  <main>
    <form
      onSubmit={[Function]}
    >
      <input
        name="name"
        onChange={[Function]}
        value=""
      />
      <input
        checked={true}
        name="sale"
        onChange={[Function]}
        type="checkbox"
      />
      <input
        name="price"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <div>
        <label>
          <input
            checked={false}
            name="tags"
            onChange={[Function]}
            type="checkbox"
            value="motor"
          />
          motor
        </label>
        <label>
          <input
            checked={false}
            name="tags"
            onChange={[Function]}
            type="checkbox"
            value="mobile"
          />
          mobile
        </label>
        <label>
          <input
            checked={false}
            name="tags"
            onChange={[Function]}
            type="checkbox"
            value="lifestyle"
          />
          lifestyle
        </label>
        <label>
          <input
            checked={false}
            name="tags"
            onChange={[Function]}
            type="checkbox"
            value="work"
          />
          work
        </label>
      </div>
      <input
        name="photo"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <img
        alt=""
        height="200"
        onClick={[Function]}
        src="placeholder.png"
        style={
          Object {
            "objectFit": "contain",
          }
        }
        width="200"
      />
      <button
        disabled={true}
      >
        Save
      </button>
    </form>
  </main>,
  <footer>
    @Keepcoding 2021
  </footer>,
]
`;
